alacritty/build.rs

```diff
@@ -11,9 +12,71 @@ fn main() {
     let dest = env::var("OUT_DIR").unwrap();
     let mut file = File::create(&Path::new(&dest).join("gl_bindings.rs")).unwrap();
 
-    Registry::new(Api::Gl, (4, 5), Profile::Core, Fallbacks::All, ["GL_ARB_blend_func_extended"])
-        .write_bindings(GlobalGenerator, &mut file)
-        .unwrap();
+    // A list of gl commands alacritty is using.
+    let used_gl_cmds: HashSet<&'static str> = [
+        "ActiveTexture",
+        "AttachShader",
+        "BindBuffer",
+        "BindTexture",
+        "BindVertexArray",
+        "BindVertexArray",
+        "BlendFunc",
+        "BlendFuncSeparate",
+        "BufferData",
+        "BufferSubData",
+        "Clear",
+        "ClearColor",
+        "CompileShader",
+        "CreateProgram",
+        "CreateShader",
+        "DeleteProgram",
+        "DeleteShader",
+        "DepthMask",
+        "DrawElements",
+        "DrawElementsInstanced",
+        "Enable",
+        "EnableVertexAttribArray",
+        "Finish",
+        "GenBuffers",
+        "GenTextures",
+        "GenVertexArrays",
+        "GetProgramInfoLog",
+        "GetProgramiv",
+        "GetShaderInfoLog",
+        "GetShaderiv",
+        "GetShaderSource",
+        "GetUniformLocation",
+        "LinkProgram",
+        "PixelStorei",
+        "ShaderSource",
+        "TexImage2D",
+        "TexParameteri",
+        "TexSubImage2D",
+        "Uniform1i",
+        "Uniform2f",
+        "Uniform4f",
+        "UseProgram",
+        "VertexArrays",
+        "VertexAttribDivisor",
+        "VertexAttribPointer",
+        "Viewport",
+    ]
+    .iter()
+    .cloned()
+    .collect();
```

Comment:
This is absolutely massive and will have to be changed every single time we want to use a new function. We also have no good way to remove things we don't need anymore and code completion will certainly suffer under it.

The reduction in code size seems great, but is it really worth it? How much does it improve compile times, I'd imagine the difference is basically nothing.

Reply:
The size reduction is closer to 100KiB from that. And symbols loading is 20usecs, instead of 400usecs.

Also, we touch our rendererer functions like once a year, so if someone really wants to add more things they can disable filtering.

Also, there's a docs.gl which is way better for OpenGL docs than auto-completion. 

The difference is like 3-4 seconds for debug build on my system (30 instead of 33/34).

Reply:
Those arguments aren't sufficient for me to justify the development overhead introduced by this addition. I never want to put us into a situation where certain parts of our code are unmaintainable for marginal compile time improvements of the rest of it.

Reply:
I mean, it's -150KiB in binary size from just removing dead code. I'm not sure how things are unmaintainable, you mean, if we remove something and won't update this list? It could be the case where we'd need automation...

We may still reduce code size by 70KiB from just using 3.3 profile iirc(it'll be like 11K LoC).

Reply:
I don't see how this would be sustainable without automation. Just me reviewing this PR is a great example. The only way for me to properly verify that there are no unnecessary imports is to try and remove every single one and then verify that compilation fails. That's just a massive amount of time that has to be invested just to check for something like this.

If we assume that this reduces the size by 80KiB over just using 3.3 profile, that's only a 1.3% reduction in binary size. Fairly insignificant. Even if we count the whole 150KiB that's only 2.5%. If I can get that size reduction for free, I'm all for it. But taking a hit on maintainability for a 2.5% reduction in binary size? That doesn't seem worth it.